### ISSUE TITLE
Repeating BlockingIOError: [Errno 11] Resource temporarily unavailable; unusably slow connecction

### ISSUE THREAD

**Issue by ohle** [Xpra-org/xpra#4049]
**Describe the bug**
Trying to start any remote application via ssh results in the following constantly repeated error:

```
BlockingIOError: [Errno 11] Resource temporarily unavailable
Exception ignored when trying to write to the signal wakeup fd:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/xpra/child_reaper.py", line 143, in sigchld
    def sigchld(self, signum, frame) -> None:

```

The window does appear, but its contents do not update, and moving or resizing the window is extremely slow.

**To Reproduce**
`xpra start ssh://user@host/ --start xterm`

**System Information (please complete the following information):**
 - Server OS: Ubuntu 22.04 Jammy
 - Client OS: Ubuntu 23.10 Mantic
 - Xpra Server Version v5.0.3-r1
 - Xpra Client Version v5.0.3-r0

**Comment by totaam** [comment]
Something is wrong with that system.
The signal handler does very little and should not be blocking.

**Comment by ohle** [comment]
It seems to be connected to [this python bug](https://bugs.python.org/issue21595), which is apparently triggered by too many short-lived subprocesses overwhelming the python runtime's SIGCHLD handler.

I've tried a little debugging myself but didn't get very far. Any idea what could cause lots of subprocesses to be created quickly?

**Comment by totaam** [comment]
It's not clear to me which end is hitting `BlockingIOError: [Errno 11] Resource temporarily unavailable`.
Is it client-side or server-side?
If client-side, try running with `XPRA_SIGNAL_WATCHER=0 xpra ...`
If server side, please post `xpra info` for this session.

**Comment by totaam** [comment]
Maybe the commit above will help? Can you try it?

**Comment by totaam** [comment]
Not heard back, likely fixed in 5.0.4
